torchgeo/datasets/nasa_marine_debris.py

```diff
@@ -0,0 +1,385 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""NASA Marine Debris dataset."""
+
+import os
+from typing import Any, Callable, Dict, List, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pytorch_lightning as pl
+import rasterio
+import torch
+from torch import Tensor
+from torch.utils.data import DataLoader
+from torchvision.transforms import Compose
+from torchvision.utils import draw_bounding_boxes
+
+from .geo import VisionDataset
+from .utils import dataset_split, download_radiant_mlhub_dataset, extract_archive
+
+# https://github.com/pytorch/pytorch/issues/60979
+# https://github.com/pytorch/pytorch/pull/61045
+DataLoader.__module__ = "torch.utils.data"
+
+
+def collate_fn(batch: List[Dict[str, Tensor]]) -> Dict[str, Any]:
```

Comment:
I think this is equivalent to and can be replaced by `torchgeo.datasets.utils.stack_samples`

Reply:
It's not. `stack_samples` just stacks each samples images, masks, etc. in a batch together. Each image in this dataset can have a variable number of boxes so you can't stack them together (this is common for other object detection datasets as well). See my notes in the description. We could probably make a custom `stack_box_samples` or something in the future to reuse.